request.go

```diff
@@ -38,11 +40,22 @@ type MessageImprint struct {
 }
 
 // Equal compares if m and n are the same MessageImprint
+//
+// Reference: RFC 3161 2.4.2
 func (m MessageImprint) Equal(n MessageImprint) bool {
 	return m.HashAlgorithm.Algorithm.Equal(n.HashAlgorithm.Algorithm) &&
+		rawvalue.Equal(m.HashAlgorithm.Parameters, n.HashAlgorithm.Parameters) &&
 		bytes.Equal(m.HashedMessage, n.HashedMessage)
 }
 
+// ASN1NullRawValue represents the valid struct of asn1.NullRawValue
+//
+// https://pkg.go.dev/encoding/asn1#NullRawValue
+var ASN1NullRawValue = asn1.RawValue{
+	Tag:       asn1.TagNull,
+	FullBytes: []byte{asn1.TagNull, 0},
+}
```

Comment:
I see the difference between this one and the built-in one. The major difference is the `FullBytes`. The built-in one is valid but just not suitable for our scenarios.

Reply:
I don't see a reason to export this variable.

Reply:
> I see the difference between this one and the built-in one. The major difference is the `FullBytes`. The built-in one is valid but just not suitable for our scenarios.

The built-in one is `not valid`. The `FullBytes` field by definition is `tag and length`, when tag is `TagNull`, it should be `[]byte{TagNull, 0}`.

Made the variable unexported.